src/coreComponents/physicsSolvers/fluidFlow/CompositionalMultiphaseBase.cpp

```diff
@@ -151,10 +147,21 @@ void CompositionalMultiphaseFlow::RegisterDataOnMesh( Group * const MeshBodies )
         setRestartFlags( RestartFlags::NO_WRITE );
 
       elementSubRegion.registerWrapper< array2d< real64 > >( viewKeyStruct::phaseVolumeFractionOldString );
+      elementSubRegion.registerWrapper< array1d< real64 > >( viewKeyStruct::totalDensityOldString );
       elementSubRegion.registerWrapper< array2d< real64 > >( viewKeyStruct::phaseDensityOldString );
+      elementSubRegion.registerWrapper< array2d< real64 > >( viewKeyStruct::phaseMobilityOldString );
       elementSubRegion.registerWrapper< array3d< real64 > >( viewKeyStruct::phaseComponentFractionOldString );
       elementSubRegion.registerWrapper< array1d< real64 > >( viewKeyStruct::porosityOldString );
     } );
+
+    FaceManager * const faceManager = meshLevel.getFaceManager();
+    {
+      faceManager->registerWrapper< array1d< real64 > >( viewKeyStruct::facePressureString )->
+        setPlotLevel( PlotLevel::LEVEL_0 )->
+        setRegisteringObjects( this->getName() )->
+        setDescription( "An array that holds the pressures at the faces." );
+    }
```

Comment:
I think it would be useful to have face-based boundary condition for the classical FVM compositional solver as well. For instance, it would be convenient to have this face-based BC for a rigorous refinement study or a scalability test. That being said, I don't really realize how much work it would take to implement them for the compositional case. I can have look into that.

Reply:
I think that to have face b.c. for the compositional solver and use them for refinement studies we would have to specify all other fields as well (i.e., faceCompositions and eventually faceTemperature). I do agree that it could be useful though.